javarosa/src/test/java/org/javarosa/xpath/test/XPathEvalTest.java

```diff
@@ -172,6 +172,14 @@ public void doTests() {
         testEval("number(date('1969-12-31'))", null, null, new Double(-1.0));
         testEval("number(date('2008-09-05'))", null, null, new Double(14127.0));
         testEval("number(date('1941-12-07'))", null, null, new Double(-10252.0));
+        testEval("number('1970-01-01')", null, null, new Double(0.0));
+        testEval("number('1970-01-02')", null, null, new Double(1.0));
+        testEval("number('1969-12-31')", null, null, new Double(-1.0));
+        testEval("number('2008-09-05')", null, null, new Double(14127.0));
+        testEval("number('1941-12-07')", null, null, new Double(-10252.0));
+        testEval("number('1970-01')", null, null, new Double(Double.NaN));
```

Comment:
I'm curious, what happens when we try to cast over a "date looking" date that just doesn't exist, like "2016-13-13" or something? (Just trying to think of areas where we might end up with uncaptured exceptions)